### PR TITLE
module 9 solution

### PR DESCRIPTION
**Pull request by SumedhaJoshSoft**
Main.go file

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+type httpChecker struct {
+}
+
+/*
+Function having context working as client to fetch status of websit
+*/
+
+func (h httpChecker) Check(ctx context.Context, sitename string) (status bool,
+	err error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```
Comment (pari-27) [inline main.go]: why do we need this ?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+			for site := range websitesMap {
+				status, err := httpChecker.Check(ctxAsClient, site)
+				if err != nil || !status {
+					websitesMap[site] = "DOWN"
+				} else {
+					websitesMap[site] = "UP"
+				}
+			}
+		case <-ctxAsClient.Done():
+			err := ctxAsClient.Err()
```
Comment (pari-27) [inline main.go:70]: What happens if the request takes more than 1 min ?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+var websitesMap = make(map[string]string)
+
+func main() {
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go checkSites()
+
+	http.HandleFunc("/", defaultHandler)
+	http.HandleFunc("/websites", websitesHandler)
```
Comment (pari-27) [inline main.go]: use different handler for get and post. Use get and post in HandleFunc

File: main.go (hunk 1 above)
Comment (SumedhaJoshSoft) [reply main.go]: It's a timeout to wait for any website upto 1 minute only.
If it doesn't send response then we can mark it as DOWN.

File: main.go (hunk 2 above)
Comment (SumedhaJoshSoft) [reply main.go:70]: If it takes more than a minute then we are marking it as done.
It's not in a requirements, but I have added it to clear understanding on context.

File: main.go (hunk 3 above)
Comment (SumedhaJoshSoft) [reply main.go]: Sure.

**Review by pari-27 — APPROVED** [review]

**Pull request merged**